optimum/commands/export/openvino.py

```diff
@@ -305,3 +357,25 @@ def run(self):
                 library_name=library_name,
                 # **input_shapes,
             )
+
+            if quantize_after_export:
+                try:
+                    from optimum.intel import OVModelForCausalLM, OVQuantizer
+
+                    ov_config.dtype = original_dtype_value
+                    model = OVModelForCausalLM.from_pretrained(
+                        self.args.output, trust_remote_code=self.args.trust_remote_code
+                    )
+                    quantizer = OVQuantizer(model)
+                    quantization_config.tokenizer = quantization_config.tokenizer or str(self.args.output)
+                    # TODO: set save_directory=self.args.output once OV is updated to 2024.3
+                    quantizer.quantize(ov_config=OVConfig(quantization_config=quantization_config))
+                    with tempfile.TemporaryDirectory() as temp_dir:
+                        model.save_pretrained(temp_dir)
+                        ov_config.save_pretrained(self.args.output)
+                        shutil.copy(f"{temp_dir}/openvino_model.xml", f"{self.args.output}/openvino_model.xml")
+                        shutil.copy(f"{temp_dir}/openvino_model.bin", f"{self.args.output}/openvino_model.bin")
+                except Exception as e:
+                    # Delete non-compressed model if compression failed for some reason
+                    shutil.rmtree(str(self.args.output))
+                    raise e
```

Comment:
Why not exporting + applying quantization using `OVModelForCausalLM` directly (and not calling `main_export` for this specific case) ?

Reply:
If possible then it could actually make sense to do this for all models (as it's already the case for SD models)

Reply:
> Why not exporting + applying quantization using OVModelForCausalLM directly (and not calling main_export for this specific case) ?

Thanks for the suggestion! This is indeed better.

> If possible then it could actually make sense to do this for all models (as it's already the case for SD models)

It would be more convenient from code maintenance side. But compared to calling `main_export()` with provided `quantization_config`, when we export and quantize through `.from_pretrained()` there is an additional model saving step to temporary directory which leads to an additional overhead, especially for large models. I think we should if avoid this if possible.